huaweicloud/services/cbr/data_source_huaweicloud_cbr_external_vaults.go

```diff
@@ -0,0 +1,522 @@
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"exclude_volumes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func dataExternalVaultBillingSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"allocated": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"used": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"charging_mode": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cloud_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"consistent_level": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"protect_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"object_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"spec_code": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"order_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"product_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"storage_unit": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"frozen_scene": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"is_multi_az": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildQueryParams(d *schema.ResourceData) string {
+	queryParams := ""
+	if v, ok := d.GetOk("external_project_id"); ok {
+		queryParams = fmt.Sprintf("%s&external_project_id=%v", queryParams, v)
+	}
+	if v, ok := d.GetOk("region_id"); ok {
+		queryParams = fmt.Sprintf("%s&region_id=%v", queryParams, v)
+	}
+	if v, ok := d.GetOk("cloud_type"); ok {
+		queryParams = fmt.Sprintf("%s&cloud_type=%v", queryParams, v)
+	}
+	if v, ok := d.GetOk("object_type"); ok {
+		queryParams = fmt.Sprintf("%s&object_type=%v", queryParams, v)
+	}
+	if v, ok := d.GetOk("protect_type"); ok {
+		queryParams = fmt.Sprintf("%s&protect_type=%v", queryParams, v)
+	}
+	if v, ok := d.GetOk("vault_id"); ok {
+		queryParams = fmt.Sprintf("%s&vault_id=%v", queryParams, v)
+	}
+
+	return queryParams
+}
+
+func dataSourceExternalVaultsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v3/{project_id}/vaults/external"
+		product = "cbr"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating CBR client: %s", err)
+	}
+
+	requestPath := client.Endpoint + httpUrl
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+
+	queryParams := buildQueryParams(d)
+	if len(queryParams) > 0 {
+		requestPath = requestPath + "?" + queryParams
```

Comment:
What about "&" character?   Seems like this query param style (?&xxx=xxx) is uncorrect.